app/ui/resources.tsx

```diff
@@ -106,30 +117,104 @@ export function InitCodeblock({
         </code>
       </pre>
 
-      <button
-        type="button"
-        onClick={() => {
-          setCopied(true);
-          navigator.clipboard.writeText(initCommand);
-        }}
-        data-code-block-copy
+      <CopyCodeBlock setCopied={handleCopied} copied={copied} />
+    </div>
+  );
+}
+
+type CopyCodeBlockProps = {
+  copied: boolean;
+  setCopied: (copied: boolean, packageManager: PackageManager) => void;
+};
+
+function CopyCodeBlock({ copied, setCopied }: CopyCodeBlockProps) {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  return (
+    <DetailsMenu
+      className="absolute right-4 top-0 !opacity-100"
+      data-copied={copied}
+      onToggle={() => setIsMenuOpen((oldValue) => !oldValue)}
+    >
+      <summary
+        className="_no-triangle absolute top-0 grid"
         data-copied={copied}
-        className="outline-none"
       >
-        {/* had to put these here instead of as a mask so we could add an opacity */}
-        <svg
-          aria-hidden
-          className="h-5 w-5 text-gray-500 hover:text-black dark:text-gray-400 dark:hover:text-gray-100"
-          viewBox="0 0 24 24"
+        <span
+          data-code-block-copy
+          data-copied={copied}
+          className={`absolute right-0 top-0 opacity-0 hover:opacity-100 ${copied || isMenuOpen ? "!opacity-100" : ""}`}
         >
-          {copied ? (
-            <use href={`${iconsHref}#check-mark`} />
-          ) : (
-            <use href={`${iconsHref}#copy`} />
-          )}
-        </svg>
-        <span className="sr-only">Copy code to clipboard</span>
-      </button>
-    </div>
+          <svg
+            aria-hidden
+            className="h-5 w-5 text-gray-500 hover:text-black dark:text-gray-400 dark:hover:text-gray-100"
+            viewBox="0 0 24 24"
+          >
+            {copied ? (
+              <use href={`${iconsHref}#check-mark`} />
+            ) : (
+              <use href={`${iconsHref}#copy`} />
+            )}
+          </svg>
+          <span className="sr-only">Copy code to clipboard</span>
+        </span>
+      </summary>
+      <DetailsPopup
+        className="!left-auto !right-0 top-10"
+        childrenClassName="!w-[110px]"
```

Comment:
I don't really love how much css with `!important` we're throwing around, this really shouldn't be necessary. For now you can leave it and make the other suggestions I made and I'll try to look into a better suggestion. Basically you should be able to adjust the parent styles to get this to work I'm pretty sure